R/calculateDMM.R

```diff
@@ -77,28 +77,19 @@
 #' tse <- TreeSummarizedExperiment(assays = list(counts = counts),
 #'                                 colData = colData)
 #'
+#' library(bluster)
+#' 
+#' # Compute DMM algorithm and store result in metadata
+#' tse <- cluster(tse, name = "DMM", DmmParam(),
+#'                MARGIN = "samples", full = TRUE)
+#' 
+#' # Get the list of DMN objects
+#' metadata(tse)$DMM$dmm
+#' # Get which objects fits best
+#' metadata(tse)$DMM$best
+#' # Get the model that fits best
```

Comment:
-> "Get the model that generated the best fit (i.e. this is the model used for the returned clustering)"

Reply:
A main purpose for having access to the model is that it contains cluster assignment probabilities for the samples. That could be used more, if we do not include wrapper for that immediately, I would at least show in the examples how to fetch the sample-cluster assignment probability matrix from the model.

Reply:
Another use if to characterize the components. The examples could also show how to fetch the DMM parameters from the model, in particular the cluster "centroids" (though they are not called centroids in this context I think but instead cluster profiles or something).

Reply:
To simplify readability, those exapmples could first fetch the single top model, then show how to do these two operations (cluster assignments and cluster profiles) for that top model.

Reply:
I don't to to which parameters you are referring to. The parameters accessible from `DMMParam` are: best (the optimal number of clusters), dmm, (seed) and prob (the probability matrix).
I could create a function to compute the centroids or cluster mean, but I don't really know what you mean by cluster profile.

Reply:
"prob" is the sample-cluster assignment matrix I think. The probabilisties there should some up to 100% per sample. Good to show how to fetch that matrix for the top model.

The cluster profiles mean the cluster-specific Dirichlet probabilities. I think these are included in the dmm (model).

Reply:
Ok, thanks. I updated the examples to show how to:

- Get the probabilities matrix for the best fit
- Get and display the dmm model that generated the best fit
- Get the weight of each component for the dmm model that generated the best fit

What do you think?

Reply:
Pefect.